### PR TITLE
Simpler solution #47280

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket:  https://core.trac.wordpress.org/ticket/47280

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-comment-query.php [hunk 1]
```diff
@@ -992,6 +988,12 @@ private function set_found_comments() {
 		global $wpdb;
 
 		if ( $this->query_vars['number'] && ! $this->query_vars['no_found_rows'] ) {
+			$found_request = "SELECT COUNT(*)
+			{$this->sql_clauses['from']}
+			{$this->sql_clauses['where']}
+			{$this->sql_clauses['groupby']}
+			{$this->sql_clauses['orderby']}
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-comment-query.php:995]: For count value from the quer,y i don't think we need `groupby` and `orderby`. Is there any use case in which we need these clauses?

File: src/wp-includes/class-wp-network-query.php [hunk 2]
```diff
@@ -510,6 +506,12 @@ private function set_found_networks() {
 		global $wpdb;
 
 		if ( $this->query_vars['number'] && ! $this->query_vars['no_found_rows'] ) {
+			$found_request = "SELECT COUNT(*)
+			{$this->sql_clauses['from']}
+			{$this->sql_clauses['where']}
+			{$this->sql_clauses['groupby']}
+			{$this->sql_clauses['orderby']}
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-network-query.php:513]: Please check above review comment

File: src/wp-includes/class-wp-query.php [hunk 3]
```diff
@@ -3014,13 +3014,10 @@ public function get_posts() {
 			$orderby = 'ORDER BY ' . $orderby;
 		}
 
-		$found_rows = '';
-		if ( ! $q['no_found_rows'] && ! empty( $limits ) ) {
-			$found_rows = 'SQL_CALC_FOUND_ROWS';
-		}
+
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-query.php:3017]: There is one empty line that needs to remove.

File: src/wp-includes/class-wp-user-query.php [hunk 4]
```diff
@@ -815,6 +811,11 @@ public function query() {
 			}
 
 			if ( isset( $qv['count_total'] ) && $qv['count_total'] ) {
+				$found_request = "SELECT COUNT(*)
+				{$this->query_from}
+				{$this->query_where}
+				{$this->query_orderby}
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-user-query.php:817]: Check my first review comment

File: src/wp-includes/class-wp-site-query.php [hunk 5]
```diff
@@ -720,6 +721,14 @@ private function set_found_sites() {
 		global $wpdb;
 
 		if ( $this->query_vars['number'] && ! $this->query_vars['no_found_rows'] ) {
+
+			$found_request = "SELECT COUNT(*)
+			{$this->sql_clauses['from']}
+			{$this->sql_clauses['where']}
+			{$this->sql_clauses['groupby']}
+			{$this->sql_clauses['orderby']}
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-site-query.php:729]: Check my first review comment.

File: src/wp-includes/class-wp-query.php [hunk 6]
```diff
@@ -3114,14 +3119,22 @@ public function get_posts() {
...
 					$orderby
 					$limits
 				";
 
+				$this->request_found = "
+					SELECT COUNT(*)
+					FROM {$wpdb->posts} $join
+					WHERE 1=1 $where
+					$groupby
+					$orderby
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-query.php:3135]: Check my first review comment.

File: src/wp-includes/class-wp-query.php [hunk 7]
```diff
@@ -3030,6 +3027,14 @@ public function get_posts() {
 
 		$this->request = $old_request;
 
+		$this->request_found = "
+					SELECT COUNT(*)
+					FROM {$wpdb->posts} $join
+					WHERE 1=1 $where
+					$groupby
+					$orderby
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-query.php:3035]: Check my first review comment.

File: src/wp-includes/class-wp-site-query.php [hunk 8]
```diff
@@ -728,7 +737,7 @@ private function set_found_sites() {
 			 * @param string        $found_sites_query SQL query. Default 'SELECT FOUND_ROWS()'.
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-site-query.php:737]: Needs to update docblock regarding new update

File: src/wp-includes/class-wp-user-query.php [hunk 9]
```diff
@@ -826,7 +827,7 @@ public function query() {
 				 * @param string        $sql   The SELECT FOUND_ROWS() query for the current WP_User_Query.
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-user-query.php:827]: Needs to update docblock regarding new update

**Review by mukeshpanchal27 — COMMENTED** [review]
The PR looks good to me. I have added some minor reviews on PR.

**Comment by MahdiAkrami01** [comment]
`COUNT(*)` query is not working when using `GROUP BY` in sql query . 
and if you remove `GROUP BY`the result is not the same as `SELECT FOUND_ROWS()`

see this exam. main query result has 415 rows but with count query if we remove `GROUP BY` we get 423 rows

image
image

**Pull request closed without merging**